### PR TITLE
Move StreamingProxyRedirects to GA

### PR DESCRIPTION
**Pull request by timstclair**
Justification for moving to GA:

- Has been in Beta for a release (even though the comments incorrectly say alpha), and alpha a full release before that, with no major issues
- Similar logic has been added to the SpdyRoundTripper, and isn't behind a feature flag (https://github.com/kubernetes/kubernetes/pull/44451)
- Features should naturally progress to GA, unless there's a specific reason not to (i.e. this is a cleanup)

```release-note
Graduate the `StreamingProxyRedirects` feature to GA.
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/features/kube_features.go [hunk 1]
```diff
@@ -43,5 +43,5 @@ func init() {
 // To add a new feature, define a key for it above and add it here. The features will be
 // available throughout Kubernetes binaries.
 var defaultKubernetesFeatureGates = map[utilfeature.Feature]utilfeature.FeatureSpec{
-	StreamingProxyRedirects: {Default: true, PreRelease: utilfeature.Beta},
+	StreamingProxyRedirects: {Default: true, PreRelease: utilfeature.GA},
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/features/kube_features.go]: is this gate honored anywhere now?

Comment (cblecker) [inline staging/src/k8s.io/apiserver/pkg/features/kube_features.go]: I searched the codebase, and can't find any references to it other than here, line 35 above, and line 106 of `pkg/features/kube_features.go`

Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/features/kube_features.go]: soo.... should it be removed?

Comment (timstclair) [inline staging/src/k8s.io/apiserver/pkg/features/kube_features.go]: If we completely remove it, it means that a cluster that was specifying the feature flag will break upon upgrade (without changing the flag): https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go#L124-L127

Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/features/kube_features.go]: ah. still seems deceptive to silently accept --feature-gates=StreamingProxyRedirects=false and stream them anyway.

Comment (timstclair) [inline staging/src/k8s.io/apiserver/pkg/features/kube_features.go]: Fair point. The only precedent I see for features graduating to GA is `ExternalTrafficLocalOnly` which flipped to GA, but left the flag checks in. I can do that here, but I assume we're going to want to clean these up eventually. Maybe specifying a flag for a GA feature should trigger a warning message that states that the feature gate will be removed in a future release?

/cc @jlowdermilk

**Comment by liggitt** [comment]
graduating to GA LGTM, what's the plan for resolving the question around flag gate behavior around GA flags?

**Comment by cblecker** [comment]
I think https://github.com/kubernetes/kubernetes/issues/46404 is relevant

**Comment by grodrigues3** [comment]
Looks like this feature wasn't approved and lgtm'd prior to code freeze.  Removing the 1.7 milestone.

**Pull request closed without merging**